compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -1891,6 +1929,114 @@ SmallVector<Range> UnPackOp::getIterationDomain(OpBuilder &builder) {
   return LinalgExt::getIterationDomain(*this, builder);
 }
 
+//===----------------------------------------------------------------------===//
+// ExpReductionOp
+//===----------------------------------------------------------------------===//
+
+SmallVector<utils::IteratorType> ExpReductionOp::getLoopIteratorTypes() {
+  SmallVector<utils::IteratorType> iteratorTypes(
+      getIteratorTypes()
+          .getAsValueRange<IREE::LinalgExt::IteratorTypeAttr,
+                           utils::IteratorType>());
+  return iteratorTypes;
+}
+
+SmallVector<Range> ExpReductionOp::getIterationDomain(OpBuilder &b) {
+  ExpReductionOp op = *this;
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(op);
+  Location loc = getLoc();
+  SmallVector<OpFoldResult> allShapesSizes =
+      createFlatListOfOperandDims(b, loc, op);
+  AffineMap map = getShapesToLoopsMap();
+
+  return llvm::map_to_vector(map.getResults(), [&](AffineExpr loopExpr) {
+    OpFoldResult ofr =
+        affine::makeComposedFoldedAffineApply(b, loc, loopExpr, allShapesSizes);
+    return Range{b.getIndexAttr(0), ofr, b.getIndexAttr(1)};
+  });
+}
```

Comment:
1. I don't think you need to set insertion point.
2. Can the implementation be easier given that they are projected permutation now?